Files add stdOut options

bmordan opened the pull request:
This PR adds the following options to `add`

```sh
--quiet
--quieter
--silent
```

### Quiet [bool]

Just log out the output.Hash (leave out the output.HashName) Each entry on a new line


### Quieter [bool]

Like quiet only just log out the final line/file


### Silent [bool]

Don’t write anything to stdout

In preparation for extending the options that the add command will accept the tests for add, cat and get are pulled into their own test files.

The previous implementation of the `--quiet` flag suppressed all the messages. This has been removed as it is not the way **go-ipfs** treats the `--quiet` flag.

bmordan committed:
updates cli to use --quiet flag correctly

bmordan committed:
Extends the 'ipfs add' options

In preparation for extending the options that the add command will accept the tests for add, cat and get are pulled into their own test files. The previous implementation of the -q quiet flag has be updated to have the same behaviour as the --quiet flag in go-ipfs

bmordan committed:
Adds the --silent flag for add

bmordan committed:
Update add.js

alanshaw commented:
👓

alanshaw commented on src/cli/bin.js:
```diff
@@ -15,12 +15,6 @@ updateNotifier({
-    desc: 'suppress output',
-    type: 'boolean',
-    coerce: (quiet) => { if (quiet) { utils.disablePrinting() } }
-  })
```
This removes this option for **all** commands. We need to find a way of removing it for just the ipfs.files.add command.

alanshaw commented on src/cli/commands/files/add.js:
```diff
@@ -69,16 +78,24 @@ function addPipeline (index, addStream, list, wrapWithDirectory) {
         })
-        .forEach((msg) => print(msg))
+
+      if (quieter) return print(messages.pop().replace('added ', ''))
```
Messages are sorted and reversed above, so this isn't the last one

alanshaw commented on test/cli/general.js:
```diff
@@ -1,13 +0,0 @@
-      expect(out).to.be.empty()
-    })
-  })
-}))
```
This should still work!

bmordan committed:
Changes global --quiet flag to be --silent

This change swaps out the global quiet flag for the --silent flag. Things were getting confusing because the files add version of quiet is not total cli silence, but a more essencial form of logging.

bmordan committed:
partial resolve of merge conflicts
